src/Semantic/Task/Files.hs

```diff
@@ -92,14 +101,9 @@ data FilesArg
   | FilesFromPaths [File]
 
 -- | A task which reads a list of 'Blob's from a 'Handle' or a list of 'FilePath's optionally paired with 'Language's.
-readBlobs :: (Has Files sig m, MonadIO m) => FilesArg -> m [Blob]
+readBlobs :: Has Files sig m => FilesArg -> m [Blob]
 readBlobs (FilesFromHandle handle) = send (Read (FromHandle handle) pure)
-readBlobs (FilesFromPaths [path]) = do
-  isDir <- isDirectory (filePath path)
```

Comment:
This cleans up some weird semantics where we were letting directory paths through and doing specious try-to-do-the-right-thing gyrations. It’s better to just reject special cases like this anyway, since our internal systems no longer hit this code path.

Reply:
I wonder if we should have a `-r` flag for recursive-mode to recover the other behaviour (with a separate code path, natch).